mesonbuild/backend/ninjabackend.py

```diff
@@ -434,18 +434,23 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
         self.introspection_data = {}
         self.created_llvm_ir_rule = PerMachine(False, False)
 
-    def create_target_alias(self, to_target):
-        # We need to use aliases for targets that might be used as directory
-        # names to workaround a Ninja bug that breaks `ninja -t clean`.
-        # This is used for 'reserved' targets such as 'test', 'install',
-        # 'benchmark', etc, and also for RunTargets.
-        # https://github.com/mesonbuild/meson/issues/1644
-        if not to_target.startswith('meson-'):
-            raise AssertionError(f'Invalid usage of create_target_alias with {to_target!r}')
-        from_target = to_target[len('meson-'):]
-        elem = NinjaBuildElement(self.all_outputs, from_target, 'phony', to_target)
+    def create_phony_target(self, all_outputs, dummy_outfile, rulename, phony_infilename, implicit_outs=None):
```

Comment:
is there actually a case where `phony_infilename` isn't going to be `'PHONY'`?

Reply:
I did this to have the same API for both `create_phony_target` and `NinjaBuildElement`, but it's probably nonsensical to use it for anything other than "phony", yes.

Ultimately this hack feels terrible, ninja is flawed. Someday I'd like to drop this function and just use NinjaBuildElement here...